index.js

```diff
@@ -24,45 +35,62 @@ module.exports = async function loader(source) {
     const contractsBuildDirectory = config.contracts_build_directory;
     const contractFileName = path.basename(contractFilePath);
     const contractName = params.contract || contractFileName.charAt(0).toUpperCase() + contractFileName.slice(1, contractFileName.length - 4);
-    const compiledContractPath = path
-      .resolve(contractsBuildDirectory, `${contractName}.json`);
+    const compiledContractPath = path.resolve(contractsBuildDirectory, `${contractName}.json`);
 
     // if loader is disabled do not compile/push/upgrade, but still serve .json contracts from file system.
     if (!disabled) {
-      // wait until compile/push/update is done
-      while (isZeppelinBusy) await wait(500);
+      // check if local version is installed
+      const isLocal = await packageExist(oz, cwd);
+      // check if global version is installed
+      const isGlobal = !!which.sync(oz, { nothrow: true });
```

Comment:
Let's avoid running this if `isLocal` is already true, since `which` [seems a bit expensive](https://github.com/npm/node-which/blob/master/which.js) to execute.